scanpy/preprocessing/_highly_variable_genes.py

```diff
@@ -57,10 +57,10 @@ def _highly_variable_genes_seurat_v3(
         )
 
     X = adata.layers[layer] if layer is not None else adata.X
-    if check_nonnegative_integers(X) is False:
-        raise ValueError(
-            "`pp.highly_variable_genes` with `flavor='seurat_v3'` expects "
-            "raw count data."
+    if check_values:
+        warnings.warn(
+            "`flavor='seurat_v3'` expects raw count data, but non-integers were found.",
+            UserWarning,
```

Comment:
I think a bug has been introduced @ivirshup @giovp 

1. The call to `check_nonnegative_integers(X)` was removed
2. The warning should only occur when `check_values` is False
3. The call to `check_nonnegative_integers(X)` should *not* be made when `check_values` is False

Reply:
On another note, in this current state `check_nonnegative_integers` is an unused import. Do you guys check for this with flake8?